api/src/main/java/com/datatorrent/api/ClusterMode.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.api;
+
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ *
+ */
+public abstract class ClusterMode<H extends Launcher.AppHandle> extends Launcher<H>
+{
+
+  public static final Attribute<String> LIB_JARS = new Attribute<String>(new StringCodec.String2String());
+  public static final Attribute<String> ORIGINAL_APP_ID = new Attribute<String>(new StringCodec.String2String());
+  public static final Attribute<String> QUEUE_NAME = new Attribute<String>(new StringCodec.String2String());
```

Comment:
Do not view them as attributes, they are simply parameters that need to be passed in and they can be different types such as ints, strings, boolean are other objects, attribute object neatly fits that bill  to pass parameters of different types. I can simply extend attribute and call it parameter and it would give me the same functionality.